lib/queue_config.rb

```diff
@@ -0,0 +1,41 @@
+require_relative './random_tools.rb'
+
+module Upaya
+  module QueueConfig
+    # rubocop:disable Metrics/MethodLength
+
+    # Known acceptable values for config.active_job.queue_adapter
+    KNOWN_QUEUE_ADAPTERS = %i[sidekiq inline async].freeze
+
+    # Select a queue adapter for use, including possible random weights as
+    # defined by Figaro.env.queue_adapter_weights (a JSON mapping from queue
+    # adapters to integer weights)..
+    def self.choose_queue_adapter
+      adapter_config = Figaro.env.queue_adapter_weights
+
+      # default to Sidekiq if no config present
+      return :sidekiq unless adapter_config
+
+      options = JSON.parse(Figaro.env.queue_adapter_weights,
+                           symbolize_names: true)
+
+      options.each_key do |adapter|
+        unless KNOWN_QUEUE_ADAPTERS.include?(adapter)
```

Comment:
Rather than performing validation in here, what do you think about doing it in `lib/config_validator`?

Reply:
No strong opinion here. I tend to lean toward keeping the validation closer to the use case when the validation wouldn't benefit from reuse, but could move this.